lib/chai-exclude.js

```diff
@@ -12,12 +12,66 @@ module.exports = (chai, utils) => {
    */
   function removeKeysFrom (val, props, recursive = false) {
     if (isObject(val)) {
+
+      // Stringify all circular object references.
+      // TODO: Think of a better way to stringify circular keys.
+      const cyclicInfo = getCyclic(val)
+      cyclicInfo.circularKeys.forEach((x) => eval(`val['${x.split('.').join("']['")}'] = '[Circular]'`))
```

Comment:
I'm seeing `eval` after years. Also,`eval` is not considered safe. We might want to do something else here. Can't we mutate `val` without using `eval`?

Reply:
I was hesitant using `eval` as well. But I haven't found another way. I'm looking for an elegant way to do it but for now it works.